lib/puppet/util/yaml.rb

```diff
@@ -9,6 +9,21 @@ module Puppet::Util::Yaml
 
   class YamlLoadError < Puppet::Error; end
 
+  def self.disallowed_class(cls)
+    raise YamlLoadError, "Tried to load unspecified class: #{cls}"
+  end
+
+  if Gem::Version.new(RUBY_VERSION.dup) >= Gem::Version.new('2.1.0')
```

Comment:
Am not asking a change but was curious if we could use "if YAML.respond_to? :safe_load" or are there any downsides to it?

Reply:
We could, but in my view, relying on version is better since it aligns with documented behavior.